libs/ml/examples/bert_simple_cls_example/main.cpp

```diff
@@ -0,0 +1,307 @@
+  // 0.
+  std::cout << "FETCH BERT Toy CLS Demo" << std::endl;
+
+  SizeType train_size = 1000;
+  SizeType test_size  = 100;
+  SizeType batch_size = 16;
+  SizeType epochs     = 2;
+  DataType lr         = static_cast<DataType>(1e-3);
+
+  BERTConfig config;
+  config.n_encoder_layers  = 2u;
+  config.max_seq_len       = 20u;
+  config.model_dims        = 12u;
+  config.n_heads           = 2u;
+  config.ff_dims           = 12u;
+  config.vocab_size        = 4u;
+  config.segment_size      = 1u;
+  config.dropout_keep_prob = static_cast<DataType>(0.9);
+
+  BERTInterface interface(config);
+
+  // create custom bert model
+  GraphType g;
+  MakeBertModel(config, g);
+
+  // Add linear classification layer
+  std::string cls_token_output = g.template AddNode<fetch::ml::ops::Slice<TensorType>>(
+      "ClsTokenOutput", {interface.outputs[interface.outputs.size() - 1]}, 0u, 1u);
+  std::string classification_output =
+      g.template AddNode<fetch::ml::layers::FullyConnected<TensorType>>(
+          "ClassificationOutput", {cls_token_output}, config.model_dims, 2u,
+          ActivationType::SOFTMAX, RegType::NONE, static_cast<DataType>(0),
+          WeightsInitType::XAVIER_GLOROT, false);
+  // Set up error signal
+  std::string label = g.template AddNode<PlaceHolder<TensorType>>("Label", {});
+  std::string error =
+      g.template AddNode<CrossEntropyLoss<TensorType>>("Error", {classification_output, label});
+
+  // Do pre-validation
+  auto test_data = PrepareToyClsDataset(test_size, config, static_cast<SizeType>(1));
+  EvaluateGraph(g, interface.inputs, classification_output, test_data.first, test_data.second);
+
+  // Do training
+  auto          train_data = PrepareToyClsDataset(train_size, config, static_cast<SizeType>(0));
+  OptimiserType optimiser(std::make_shared<GraphType>(g), interface.inputs, label, error, lr);
+
+  for (SizeType i = 0; i < epochs; i++)
+  {
+    optimiser.Run(train_data.first, train_data.second, batch_size);
+    EvaluateGraph(g, interface.inputs, classification_output, test_data.first, test_data.second,
+                  false);
+  }
+
+  // Do validation
+  EvaluateGraph(g, interface.inputs, classification_output, test_data.first, test_data.second);
+
+  return 0;
+}
+
+std::pair<std::vector<std::string>, std::vector<std::string>> MakeBertModel(
+    BERTConfig const &config, GraphType &g)
+{
+  // create a bert model based on the config passed in
+  SizeType n_encoder_layers  = config.n_encoder_layers;
+  SizeType max_seq_len       = config.max_seq_len;
+  SizeType model_dims        = config.model_dims;
+  SizeType n_heads           = config.n_heads;
+  SizeType ff_dims           = config.ff_dims;
+  SizeType vocab_size        = config.vocab_size;
+  SizeType segment_size      = config.segment_size;
+  DataType epsilon           = config.epsilon;
+  DataType dropout_keep_prob = config.dropout_keep_prob;
+
+  std::cout << max_seq_len << std::endl;
```

Comment:
if this is an intentional print statement then change it to:
std::cout << "max_seq_len: " << max_seq_len << std::endl;

Reply:
no it is not